src/strategy/dungeons/wotlk/forgeofsouls/ForgeOfSoulsActions.h

```diff
@@ -0,0 +1,37 @@
+#ifndef _PLAYERBOT_WOTLKDUNGEONFOSACTIONS_H
+#define _PLAYERBOT_WOTLKDUNGEONFOSACTIONS_H
+
+#include "Action.h"
+#include "AttackAction.h"
+#include "PlayerbotAI.h"
+#include "Playerbots.h"
+#include "ForgeOfSoulsTriggers.h"
+
+const Position BRONJAHM_TANK_POSITION = Position(5297.920f, 2506.698f, 686.068f);
+
+class MoveFromBronjahmAction : public MovementAction
+{
+public:
+    MoveFromBronjahmAction(PlayerbotAI* ai) : MovementAction(ai, "move from bronjahm") {}
+    bool Execute(Event event) override;
+};
+
+class AttackCorruptedSoulFragmentAction : public AttackAction
+{
+public:
+    AttackCorruptedSoulFragmentAction(PlayerbotAI* ai) : AttackAction(ai, "attack corrupted soul fragment") {}
+    bool Execute(Event event) override;
+};
+
+class BronjahmGroupPositionAction : public AttackAction
```

Comment:
Positioning action should probably be a MovementAction, see .cpp notes

Reply:
It sounds like there may be an error in your logic checking or multiplier if this is happening.
But assuming you've tested this, we can probably merge this and get people testing it and providing feedback.

I'll look into it once merged to see where we can correct the logic flow.

Thanks again for the PR!